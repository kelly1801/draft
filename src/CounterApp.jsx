import React from "react";
import { useState } from "react";
import PropTypes from 'prop-types'

export const CounterApp = ({value}) => {
  const [count, setCount] = useState(value);
  const handleClick = (val) => {
    
    setCount((prev) => Math.max(prev + val, 0));
  };
  const resetCount = () => {
    setCount(value)
  }
  return (
    <>
      <h1>CounterApp</h1>
      <h2> {count} </h2>
      <button onClick={() => handleClick(1)}>+1</button>
      <button onClick={() => handleClick(-1)}>-1</button>
      <button aria-label="btn-reset" onClick={resetCount}>reset</button>
    </>
  );
};


CounterApp.propTypes = {
  value: PropTypes.number.isRequired
}