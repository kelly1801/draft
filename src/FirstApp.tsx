import React from "react";
import PropTypes from 'prop-types'

export const FirstApp = ({ name, subtitle }) => {
  
  return (
    <>
      <h1>Hola soy {name}</h1>
      <p>{subtitle}</p>
   
    </>
  )
};


FirstApp.propTypes = {
  name: PropTypes.string.isRequired,
  subtitle: PropTypes.string
}
FirstApp.defaultProps = {
  name: 'Santos',
  subtitle: "Subtitulo"
}