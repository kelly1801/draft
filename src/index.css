body {
  background-color: #1c0505;
  color: #f2f2f2
}
