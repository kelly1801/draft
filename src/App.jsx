import {CounterApp} from './CounterApp'
const App = () => {
 

  return (
    <>
    <CounterApp value={2}/>
    </>
  )
}

export default App
