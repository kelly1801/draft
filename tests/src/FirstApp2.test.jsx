//snapshot is like a picture of the component
import React from "react";
import {  render, screen } from "@testing-library/react";
import { FirstApp } from "../../src/FirstApp";

describe("Pruebas en <FistMap/>", () => {
    const name = "kelly"
    const subtitle = "soy un subtitulo"

    test('debe de hacer match con el snapshot', () => {
        
      const { container } = render( <FirstApp name={ name } /> );
      expect( container ).toMatchSnapshot();

  });
    test('debe de mostrar el mensaje "Hola, Soy kelly"', () => {
        
      render( <FirstApp name={ name } /> );
      expect( screen.getByText(name) ).toBeTruthy();
      
  });

  test('debe de mostrar el titulo en un h1', () => {
      render( <FirstApp name={ name } /> );
      expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( name );
  });

  test('debe de mostrar el subtitulo', () => {
    render( <FirstApp name={ name } subtitle={subtitle}/> );
    expect( screen.getAllByText(subtitle).length).toBeTruthy();
});

  
    


});
