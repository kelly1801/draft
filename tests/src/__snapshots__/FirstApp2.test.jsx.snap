// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FistMap/> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy 
    kelly
  </h1>
  <p>
    Subtitulo
  </p>
</div>
`;
